pkg/services/sqlstore/dashboard.go

```diff
@@ -473,17 +473,28 @@ func deleteDashboard(cmd *models.DeleteDashboardCommand, sess *DBSession) error
 			}
 		}
 
-		// clean ngalert tables
-		ngalertDeletes := []string{
-			"DELETE FROM alert_rule WHERE namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
-			"DELETE FROM alert_rule_version WHERE rule_namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
-		}
+		if cmd.ForceDeleteFolderRules {
+			// clean ngalert tables
+			ngalertDeletes := []string{
```

Comment:
Is this correct? `namespace_uid` and `rule_ namespace_uid` are dashboard uids?

Reply:
yes, folders are actually stored in the `dashboard` table with a flag [is_folder](https://github.com/grafana/grafana/blob/b5379c53356769e3af5a71ab06e1bafbeff10ebc/pkg/services/sqlstore/migrations/dashboard_mig.go#L146).